BetterHolidays/markets/nyse.py

```
@@ -0,0 +1,180 @@
+  def fetch_future(cls):
+    doc = md.HTML.from_url("https://www.nyse.com/markets/hours-calendars")
+    table:'elm.Table' = doc.inner_html.get_elements_by_class_name("table-data")[0]
+
+    table_dict = table.to_dict()
+    holidays = table_dict.pop("Holiday")
+
+    for year, dates in table_dict.items():
+      def handle_date(date:'str'):
+          split_date = date.split(" ")
+          
+          return dt.date(int(year), int(MONTHS_MAP[split_date[1].upper()]), int(split_date[2].replace("*", "")))
```

Ruff (0.8.2)

Function definition does not bind loop variable `year`

(B023)